### PR TITLE
Added service to get initial end efectors configuration

### PR DESCRIPTION
**Pull request by mlanas**
A service was added in ExoForce class (cage_simulation node) to allow other nodes to retrieve the initial configuration from the XML file.
The first via_point of each muscle unit (motor position)  is not included in the response, given that this changes which cage rotation and its part of the cage state, NOT initial configuration.

The cage state publisher was modified to not publish the via_points attached to the operator, it only publishes the first via_point (motor position).

### CODE REVIEW

**Comment by missxa** [comment]
> The first via_point of each muscle unit (motor position) is not included in the response, given that this changes which cage rotation and its part of the cage state, NOT initial configuration.
> 
> The cage state publisher was modified to not publish the via_points attached to the operator, it only publishes the first via_point (motor position).

i don't understand the motivation for these decisions. it will be very confusing I think. if you have to explain it now, everyone reading the code will have questions. why not publish cage state with all viapoints (it is in fact the state of the cage, cause tendons are a part of the cage)?
i don't see the reason why not include the motor position in the init config srv. it is a part of the initial state, and yea it will change and you will get the updates for that.

**Comment by mlanas** [comment]
> > The first via_point of each muscle unit (motor position) is not included in the response, given that this changes which cage rotation and its part of the cage state, NOT initial configuration.
> > The cage state publisher was modified to not publish the via_points attached to the operator, it only publishes the first via_point (motor position).
> 
> i don't understand the motivation for these decisions. it will be very confusing I think. if you have to explain it now, everyone reading the code will have questions. why not publish cage state with all viapoints (it is in fact the state of the cage, cause tendons are a part of the cage)?
> i don't see the reason why not include the motor position in the init config srv. it is a part of the initial state, and yea it will change and you will get the updates for that.

Will apply the changes

File: bulletroboy/exoforce.py [hunk 1]
```diff
@@ -248,3 +264,24 @@ def get_msg(self):
...
 	def publish_state(self):
 		self.cage_state_publisher.publish(self.get_msg())
+
+	def get_end_effectors_callback(self, request, response):
+
+		for ef in self.end_effectors:
+			end_effector_msg = EndEffector()
+			end_effector_msg.name = ef
+			for muscle in self.get_ef_muscle_units(ef):
+				muscle_msg = MuscleUnitMsg()
```
Comment (missxa) [inline bulletroboy/exoforce.py]: - [x] add `to_msg()` method in `MuscleUnit` class instead of doing it here

File: bulletroboy/exoforce.py [hunk 2]
```diff
@@ -248,3 +264,24 @@ def get_msg(self):
...
+				muscle_msg.id = muscle.id
+				muscle_msg.max_force = muscle.max_force
+				for via_point in muscle.via_points:
+					via_point_msg = ViaPoint()
```
Comment (missxa) [inline bulletroboy/exoforce.py]: same here, we could have a convenience `ViaPoint` class. wouldn't it be better?

File: bulletroboy/exoforce.py (hunk 1 above)
Comment (missxa) [reply bulletroboy/exoforce.py]: so you'd have something like
```python
for muscle in self.get_ef_muscle_units(ef):
   muscle_msg = muscle.to_msg() # viapoints to msg would be nested in that function
   end_effector_msg.muscle_units.append(muscle_msg)
```

**Review by missxa — CHANGES_REQUESTED** [review]

**Review by missxa — APPROVED** [review]

**Pull request merged**